lib/msf/core/post/linux/compile.rb

```diff
@@ -0,0 +1,64 @@
+# -*- coding: binary -*-
+require 'msf/core/post/common'
+require 'msf/core/post/file'
+require 'msf/core/post/unix'
+
+module Msf
+class Post
+module Linux
+module Compile
+  include ::Msf::Post::Common
+  include ::Msf::Post::File
+  include ::Msf::Post::Unix
+
+  def initialize(info = {})
+    super
+    register_options( [
+      OptEnum.new('COMPILE', [true, 'Compile on target', 'Auto', ['Auto', 'True', 'False']]),
+    ], self.class)
+  end
+
+  def live_compile?
+    return false unless datastore['COMPILE'].eql?('Auto') || datastore['COMPILE'].eql?('True')
+
+    if has_gcc?
+      vprint_good 'gcc is installed'
+      return true
+    end
+
+    unless datastore['COMPILE'].eql? 'Auto'
+      fail_with Failure::BadConfig, 'gcc is not installed. Set COMPILE False to upload a pre-compiled executable.'
+    end
+  end
+
+  def upload_and_compile(path, data, gcc_args='')
+    upload_binary "#{path}.c", strip_comments(data)
+
+    gcc_cmd = "gcc -o #{path} #{path}.c"
+    if session.type.eql? 'shell'
+      gcc_cmd = "PATH=$PATH:/usr/bin/ #{gcc_cmd}"
```

Comment:
This should probably be quoted.

```suggestion
      gcc_cmd = "PATH=\"$PATH:/usr/bin/\" #{gcc_cmd}"
```